ingress-controller/internal/manager/setup.go

```diff
@@ -57,6 +59,37 @@ func setupManagerOptions(ctx context.Context, logger logr.Logger, c *managercfg.
 
 	cacheOptions := cache.Options{
 		SyncPeriod: &c.SyncPeriod,
+		ByObject:   map[client.Object]cache.ByObject{},
+	}
+
+	// Add label selector for Secrets and ConfigMaps on the manager level.
+	if c.SecretLabelSelector != nil {
+		// TODO: move the code to build cache's ByObject options to common utils packages.
```

Comment:
Did not do this in the PR because moving it to common packages will make packages in `ingress-controller` to reference packages out of `ingress-controller` but in the whole project. Not sure if it will cause dependency cycles and are we ready to break the "internal wall".

Reply:
Extracted to a dedicated function in this file.